x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptor.java

```diff
@@ -541,18 +565,17 @@ public String toString() {
         public void messageReceived(T request, TransportChannel channel, Task task) {
             try (ThreadContext.StoredContext ctx = threadContext.newStoredContextPreservingResponseHeaders()) {
                 String profile = channel.getProfileName();
-                ServerTransportFilter filter = profileFilters.get(profile);
-
-                if (filter == null) {
-                    if (TransportService.DIRECT_RESPONSE_PROFILE.equals(profile)) {
-                        // apply the default filter to local requests. We never know what the request is or who sent it...
-                        filter = profileFilters.get("default");
-                    } else {
-                        String msg = "transport profile [" + profile + "] is not associated with a transport filter";
-                        throw new IllegalStateException(msg);
-                    }
-                }
+                ServerTransportFilter filter = getServerTransportFilter(profile);
```

Comment:
Had to slightly refactor this because anything used in a lambda (the log supplier below) needs to be final. I basically just extracted the below code block into `getServerTransportFilter`.